### PR TITLE
Feature/526 pagination to own projects page

### PR DESCRIPTION
**Pull request by BartvanEijkelenburg**
## Description

I have added pagination to the user projects page, also I added filters to sort the projects.

The Modal helper utility class is now being used when opening a modal.
## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [x] I updated the changelog with an end-user readable description
-   [x] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce
Make sure to have the backend Repo cloned for the Get User projects endpoint, you can find it here: https://github.com/DigitalExcellence/dex-backend/tree/feature/461-add-pagination-getuserprojects-endpoint

Run DeX, after that log in as Bob.

Go to own projects page,
as you can see there is the pagination added.

## Link to issue
https://github.com/DigitalExcellence/dex-frontend/issues/526
Closes: #526

### CODE REVIEW

File: src/app/modules/user/user-projects/user-projects.component.ts [hunk 1]
```diff
@@ -182,13 +211,13 @@ export class UserProjectsComponent implements OnInit {
       this.modalRef.setClass('project-modal');
```
Comment (waltersajtos) [inline src/app/modules/user/user-projects/user-projects.component.ts:211]: Take a look at [this helper class](https://github.com/DigitalExcellence/dex-frontend/blob/develop/src/app/utils/project-detail-modal.util.ts)

**Review by waltersajtos — CHANGES_REQUESTED** [review]

File: src/app/modules/user/user-projects/user-projects.component.ts (hunk 1 above)
Comment (BartvanEijkelenburg) [reply src/app/modules/user/user-projects/user-projects.component.ts:207]: Hi Walter, I'm wondering why I have to look at the helper class of the modal, I added pagination to the userprojects page and did not touch any of the modals

Comment (waltersajtos) [reply src/app/modules/user/user-projects/user-projects.component.ts:207]: Seems like the userProjects page is opening the modals itself, it should be using the help class instead. Would be nice if you could implement that in this PR too :) Or if @MeesvanStraten/ @DaveBouman wants to make it a different issue that is cool too

**Review by MeesvanStraten — APPROVED** [review]
Nice work!

**Comment by MeesvanStraten** [comment]
No merging untill it's backend SO is merged please :)

**Comment by DaveBouman** [comment]
Works like it should. I will force push this pull request. The reason why the sonarcloud tests do not complete is because it is a small issue and sonarcloud sees it as duplicate code while it isnt.

**Pull request merged**